File: src/main/scala/no/ovstetun/sdummy/Person.scala
package no.ovstetun.sdummy

import java.util.Date

class Person(var firstname:String, var lastname:String, var gender:Gender, var birthdate:Date)

class Person2(var firstname:String, var lastname:String, var gender:Gender, var birthdate:Option[Date] = None)

sealed trait Gender
case object Male extends Gender
case object Female extends Gender

object GenderE extends Enumeration {
  val Male, Female = Value
}


trait PersonRepository {
  def findById(id:Int) : Option[Person2] = {
    val p: Person2 = new Person2("", "", Male)
    val pO : Option[Person2] = Some(p)

    val name = pO.map(pers => pers.firstname)
    name.getOrElse("John Doe")
    pO
  }

  case class MyVal(s:String)
  def myMatch(n:Any) : String = {
    n match {
      case 1 => "ONE"
      case n:Int if n < 10 => "low..."
      case n:Int if n > 10 => "high!"
      case p:Person => "a person: " + p.firstname
      case MyVal(s) => s
    }
  }
}

File: src/main/java/no/ovstetun/jdummy/Person.java
package no.ovstetun.jdummy;

import java.util.Date;

public class Person {
    private String firstname;
    private String lastname;
    private Date birthDate;
    private Gender gender;

    public Person(String firstname, String lastname, Date birthDate, Gender gender) {
        this.firstname = firstname;
        this.lastname = lastname;
        this.birthDate = birthDate;
        this.gender = gender;
    }

    public String getFirstname() {
        return firstname;
    }
    public void setFirstname(String firstname) {
        this.firstname = firstname;
    }

    public String getLastname() {
        return lastname;
    }
    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public Date getBirthDate() {
        return birthDate;
    }
    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public Gender getGender() {
        return gender;
    }
    public void setGender(Gender gender) {
        this.gender = gender;
    }
}

interface PersonRepository {
    /**
     * @return A person if found, <code>null</code> otherwise.
     */
    Person findById(int id);
}
